client/js/_admin/reducers/accounts.js

```diff
@@ -0,0 +1,64 @@
+import _             from 'lodash';
+import { Constants } from '../actions/accounts';
+
+const initialState = {
+  accounts : {},
+  loading  : false,
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+
+    case Constants.GET_CANVAS_ACCOUNTS_DONE: {
+      const newState = _.cloneDeep(state);
+      newState.loading = false;
+      _.forEach(action.payload, (account) => {
+        newState.accounts[account.id] = account;
+      });
+
+      return newState;
+    }
+
+    case Constants.GET_CANVAS_ACCOUNTS: {
+      const newState = _.cloneDeep(state);
+      newState.loading = _.isEmpty(newState.accounts);
+
+      return newState;
+    }
+
+    case 'GET_SUB_ACCOUNTS_OF_ACCOUNT_DONE': {
```

Comment:
Accounts and subaccounts are stored nested. This is modifying the original account passed in to put the subaccounts in the right place. A shallow clone to tell any components that it is a new state an maintain object references would probably be good though.